alea/model.py

```diff
@@ -89,49 +97,53 @@ def _define_parameters(self, parameter_definition):
             raise RuntimeError("parameter_definition must be dict or list")
 
     def _check_ll_and_generate_data_signature(self):
+        """Check that the likelihood and generate_data functions have the same signature"""
         ll_params = set(inspect.signature(self._ll).parameters)
         generate_data_params = set(inspect.signature(self._generate_data).parameters)
         if ll_params != generate_data_params:
             raise AssertionError(
                 "ll and generate_data must have the same signature (parameters)")
 
     def _ll(self, **kwargs) -> float:
+        """Likelihood function, return the loglikelihood for the given parameters."""
         raise NotImplementedError(
             "You must write a likelihood function (_ll) for your statistical model"
             " or use a subclass where it is written for you")
 
     def _generate_data(self, **kwargs):
+        """Generate data for the given parameters."""
         raise NotImplementedError(
             "You must write a data-generation method (_generate_data) for your statistical model"
             " or use a subclass where it is written for you")
 
     @_needs_data
     def ll(self, **kwargs) -> float:
         """
-        Likelihod function, returns the loglikelihood for the given parameters.
+        Likelihod function, return the loglikelihood for the given parameters.
         The parameters are passed as keyword arguments, positional arguments are not possible.
         If a parameter is not given, the default value is used.
 
+        Keyword Args:
+            kwargs: keyword arguments for the parameters
+
         Returns:
-            float: Likelihood value
+            float: likelihood value
         """
         parameters = self.parameters(**kwargs)
         return self._ll(**parameters)
 
-    def generate_data(self, **kwargs):
+    def generate_data(self, **kwargs) -> dict or list:
         """
         Generate data for the given parameters.
         The parameters are passed as keyword arguments, positional arguments are not possible.
-        If a parameter is not given, the default value is used.
+        If a parameter is not given, the default values are used.
 
         Raises:
             ValueError: If the parameters are not within the fit limits
 
         Returns:
-            Data
+            dict or list: generated data
         """
-        # CAUTION:
-        # This implementation won't allow you to call generate_data by positional arguments.
```

Comment:
I wouldn't remove this. I think it's good to keep this info there for the future.

Reply:
I think here you mean:
```
This implementation won't allow you to call generate_data by arguments that are not in parameters.
```
?

Reply:
Got it.

Reply:
Added.